sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -208,7 +286,7 @@ private[client] class Shim_v0_12 extends Shim with Logging {
       predicates: Seq[Expression]): Seq[Partition] = {
     // getPartitionsByFilter() doesn't support binary comparison ops in Hive 0.12.
     // See HIVE-4888.
-    logDebug("Hive 0.12 doesn't support predicate pushdown to metastore. " +
+    logWarning("Hive 0.12 doesn't support predicate pushdown to metastore. " +
```

Comment:
For this method , it is allowed to have false positives. Let's update the doc of this method. Also, let's revert this change (just to avoid of letting users think that this warning indicates an error).